lib/csrf.js

```diff
@@ -93,14 +93,22 @@ module.exports = function (options) {
 
         if (blocklist) {
             blocklist.some(function (exclusion) {
-                shouldBypass = req.path.indexOf(exclusion) === 0;
+                if (typeof exclusion === Object && exclusion.match === 'exact') {
```

Comment:
I think it would be more optimal if we can preprocess the blocklist and allowlist, and create "Set" for 'exact', and includesBlockList and includesAllowList for the rest.